tests/EStimLibrary.UnitTests/Extensions/SpatialModel/StringHierarchy/StringHierarchyLocationFactoryTests.cs

```diff
@@ -0,0 +1,299 @@
+        // Create base region.
+        StringHierarchyRegion region = new StringHierarchyRegion("base", null!);
+
+        // Create factory with non-null base region.
+        StringHierarchyLocationFactory factory =
+            new StringHierarchyLocationFactory(region);
+
+        // Check that HelpMsg and ParamLimits are initialized correctly.
+        Assert.Equal("A StringHierarchyLocation can be built from one of " +
+            "the following path specs, selecting one option from any list " +
+            "in [] and excluding the []:\n" + region.ToString(),
+            factory.HelpMsg);
+        Assert.Single(factory.ParamLimits);
+        Assert.NotNull(factory.ParamLimits["fullSpec"]);
+    }
+
+    /// <summary>
+    /// Test the TryCreate method with paramValues not including a fullSpec
+    /// value and skipping value validation.
+    /// </summary>
+    [Fact]
+    public void TryCreate_ShouldNotValidateShouldNotCreate()
+    {
+        // Create base region
+        StringHierarchyRegion region = new StringHierarchyRegion("base", null!);
+
+        // Create factory with non-null base region
+        StringHierarchyLocationFactory factory =
+            new StringHierarchyLocationFactory(region);
+
+        // Create dictionary of parameter values
+        Dictionary<string, object> paramValues =
+            new Dictionary<string, object>();
+
+        // Create variable to store produced IArea
+        ILocation? product;
+
+        // Check TryCreate return and product
+        bool valid = factory.TryCreate(paramValues, out product, true);
+        Assert.False(valid);
+        Assert.Null(product);
+    }
+
+    /// <summary>
+    /// Test the TryCreate method with paramValues including a fullSpec value
+    /// and skipping value validation
+    /// </summary>
+    [Fact]
+    public void TryCreate_ShouldNotValidateShouldCreate()
+    {
+        // Create non-empty options, modifiers, and subregions parameters
+        var regionOptions = new HashSet<string>() { "right", "left" };
+        var regionModifiers = new Dictionary<string, HashSet<string>>
+        {
+            { "modSet1", new HashSet<string>() { "mod1", "mod2" } },
+            { "modSet2", new HashSet<string>() { "mod3", "mod4" } }
+        };
+        var regionSubregions = new Dictionary<string, StringHierarchyRegion>
+        {
+            { "child1", new StringHierarchyRegion("child1", null, null, null,
+                regionModifiers) },
+            { "child2", new StringHierarchyRegion("child2", null) }
+        };
+        // Create region with the above parameters
+        var region = new StringHierarchyRegion("base", null,
+            null, regionOptions, null, regionSubregions);
+
+        // Create factory with non-null base region
+        StringHierarchyLocationFactory factory =
+            new StringHierarchyLocationFactory(region);
+
+        // Create dictionary with valid spec
+        Dictionary<string, object> paramValues1 =
```

Comment:
Since you've written all the test cases this way already, fine for this PR, but in future, if you find yourself testing a whole bunch of different variations of data within the same test method, try to use a [Theory] tag with [InlineData()] or [MemberData()] instead